aws-logs-metricfilter/src/main/java/software/amazon/logs/metricfilter/CreateHandler.java

```diff
@@ -37,51 +29,30 @@
 
         final ResourceModel model = request.getDesiredResourceState();
 
-        // resource can auto-generate a name if not supplied by caller.
-        // this logic should move up into the CloudFormation engine, but
-        // currently exists here for backwards-compatibility with existing models
         if (StringUtils.isNullOrEmpty(model.getFilterName())) {
-            model.setFilterName(
-                    IdentifierUtils.generateResourceIdentifier(
-                            request.getLogicalResourceIdentifier(),
-                            request.getClientRequestToken(),
-                            MAX_LENGTH_METRIC_FILTER_NAME
-                    )
-            );
+            model.setFilterName(IdentifierUtils.generateResourceIdentifier(request.getLogicalResourceIdentifier(), request.getClientRequestToken(), MAX_LENGTH_METRIC_FILTER_NAME));
         }
 
         return ProgressEvent.progress(model, callbackContext)
-            .then(progress ->
-                preCreateCheck(proxy, callbackContext, proxyClient, model)
-                    .done((response) -> {
-                        if (response.metricFilters().isEmpty()) {
-                            return ProgressEvent.progress(model, callbackContext);
-                        }
-                        return ProgressEvent.defaultFailureHandler(new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, model.getPrimaryIdentifier().toString()), HandlerErrorCode.AlreadyExists);
-                    })
-            )
             .then(progress ->
                 proxy.initiate("AWS-Logs-MetricFilter::Create", proxyClient, model, callbackContext)
                     .translateToServiceRequest(Translator::translateToCreateRequest)
-                    .makeServiceCall(this::createResource)
+                    .makeServiceCall((r, c) -> createResource(model, r, c))
                     .progress())
             .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
```

Comment:
What is the ReadHandler doing here? is there any extra value by read?

Reply:
removed